bundles/org.openhab.binding.xmltv/src/main/java/org/openhab/binding/xmltv/internal/handler/ChannelHandler.java

```diff
@@ -0,0 +1,250 @@
+     *
+     * @param channelUID the id identifying the channel to be updated
+     *
+     */
+    private void updateChannel(ChannelUID channelUID) {
+        String[] uidElements = channelUID.getId().split("#");
+        if (uidElements.length == 2) {
+            int target = GROUP_NEXT_PROGRAMME.equals(uidElements[0]) ? 1 : 0;
+            if (programmes.size() > target) {
+                Programme programme = programmes.get(target);
+
+                switch (uidElements[1]) {
+                    case CHANNEL_ICON:
+                        updateState(channelUID,
+                                GROUP_CHANNEL_PROPERTIES.equals(uidElements[0])
+                                        ? mediaIcon != null ? mediaIcon : UnDefType.NULL
+                                        : programmeIcon != null ? programmeIcon : UnDefType.NULL);
+                        break;
+                    case CHANNEL_CHANNEL_URL:
+                        updateState(channelUID,
+                                mediaChannel != null ? mediaChannel.getIcons().size() > 0
+                                        ? new StringType(mediaChannel.getIcons().get(0).getSrc())
+                                        : UnDefType.NULL : UnDefType.NULL);
+                        break;
+                    case CHANNEL_PROGRAMME_START:
+                        Instant is = programme.getProgrammeStart();
+                        ZonedDateTime zds = ZonedDateTime.ofInstant(is, ZoneId.systemDefault());
+                        updateState(channelUID, new DateTimeType(zds));
+                        break;
+                    case CHANNEL_PROGRAMME_END:
+                        ZonedDateTime zde = ZonedDateTime.ofInstant(programme.getProgrammeStop(),
+                                ZoneId.systemDefault());
+                        updateState(channelUID, new DateTimeType(zde));
+                        break;
+                    case CHANNEL_PROGRAMME_TITLE:
+                        List<WithLangType> titles = programme.getTitles();
+                        updateState(channelUID,
+                                titles.size() > 0 ? new StringType(programme.getTitles().get(0).getValue())
+                                        : UnDefType.NULL);
+                        break;
+                    case CHANNEL_PROGRAMME_CATEGORY:
+                        List<WithLangType> categories = programme.getCategories();
+                        updateState(channelUID,
+                                categories.size() > 0 ? new StringType(programme.getCategories().get(0).getValue())
+                                        : UnDefType.NULL);
+                        break;
+                    case CHANNEL_PROGRAMME_ICON:
+                        List<Icon> icons = programme.getIcons();
+                        updateState(channelUID,
+                                icons.size() > 0 ? new StringType(icons.get(0).getSrc()) : UnDefType.NULL);
```

Comment:
We prefer `UnDefType.UNDEF` instead. A state should only be `NULL` if the item is not initialized yet (e.g. during the start).